### PR TITLE
ref #95: Add active flag for routing (and bar during routing)

### PR DESCRIPTION
**Pull request by UlrichKu**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#95
| License       | MIT

Adds active flag and considers during (frontend) routing.

### CODE REVIEW

File: src/CmsRoutingBundle/Bridge/Chameleon/CmsRoutingTableEditor.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?php
+
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/CmsRoutingBundle/Bridge/Chameleon/CmsRoutingTableEditor.php]: ok

File: src/CmsRoutingBundle/Event/RoutingConfigChangedEvent.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?php
+
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/CmsRoutingBundle/Event/RoutingConfigChangedEvent.php]: ok

File: src/CmsRoutingBundle/Bridge/Chameleon/Migration/Script/update-1544434916.inc.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      'position' => '270',
+      'width' => '-1',
+      'align' => 'left',
+      'callback_fnc' => '',
+      'use_callback' => '0',
+      'show_in_list' => '1',
+      'show_in_sort' => '0',
+      'cms_tbl_conf_id' => TCMSLogChange::GetTableId('pkg_cms_routing'),
+      'cms_translation_field_name' => '',
+      'id' => 'fc6f7c91-c197-fdda-869e-deb35eda1b98',
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/CmsRoutingBundle/Bridge/Chameleon/Migration/Script/update-1544434916.inc.php]: ok

File: src/CmsRoutingBundle/Event/RoutingConfigChangedEvent.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+use Symfony\Component\EventDispatcher\Event;
+
+class RoutingConfigChangedEvent extends Event
+{
+    /**
+     * @var string
+     */
+    private $routingName;
+
+    public function __construct(string $routingName)
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/CmsRoutingBundle/Event/RoutingConfigChangedEvent.php]: ok (3x)

File: src/CmsRoutingBundle/Event/RoutingConfigChangedEvent.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        $this->routingName = $routingName;
+    }
+
+    /**
+     * @return string
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/CmsRoutingBundle/Event/RoutingConfigChangedEvent.php]: ok

File: src/CmsRoutingBundle/Bridge/Chameleon/CmsRoutingTableEditor.php [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+use TCMSTableEditor;
+
+class CmsRoutingTableEditor extends TCMSTableEditor
+{
+    protected function PostSaveHook(&$oFields, &$oPostTable)
+    {
+        parent::PostSaveHook($oFields, $oPostTable);
+
+        $this->getEventDispatcher()->dispatch(
+            CoreEvents::CHANGE_ACTIVATE_ROUTING_CONFIG,
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/CmsRoutingBundle/Bridge/Chameleon/CmsRoutingTableEditor.php]: ok (name change)

ok, Tdb used

File: src/CoreBundle/Routing/ChameleonFrontendRouter.php [hunk 7]
```diff
@@ -145,6 +145,7 @@ protected function getRouterConfig()
 ORDER BY `cms_portal`.`identifier` DESC, `pkg_cms_routing`.`position` ASC
 SQL;
         $configList = TdbPkgCmsRoutingList::GetList($query);
+        $configList->AddFilterString("`pkg_cms_routing`.`active` = '1'");
```
[the remark this answers is not in the document]
Comment (UlrichKu) [reply src/CoreBundle/Routing/ChameleonFrontendRouter.php]: ok

File: src/CoreBundle/EventListener/ClearRoutingCacheListener.php [hunk 8]
```diff
@@ -29,7 +29,7 @@ public function __construct(ChameleonBaseRouter $router)
         $this->router = $router;
     }
 
-    public function clearRoutingCache(Event $event)
+    public function clearRoutingCache(RoutingConfigChangedEvent $event)
```
Comment (umulmrum) [inline src/CoreBundle/EventListener/ClearRoutingCacheListener.php]: The base Event class is used intentionally - we do not care about the concrete event as we don't need any information, so we make this listener compatible to as many events as possible.

Comment (UlrichKu) [reply src/CoreBundle/EventListener/ClearRoutingCacheListener.php]: ok

**Comment by umulmrum** [comment]
Thanks @UlrichKu

**Pull request merged**